File: frontend/teampilot/src/app/services/api/v1/model/teamMatchDTO.ts
import { TeamForTournamentDTO } from './teamForTournamentDTO';

export interface TeamMatchDTO { 

matchId?: string | null;
matchLengthInMinutes?: string | null;
matchDate?: string | null;
participatingTeams?: Array<TeamForTournamentDTO> | null;

tournamentId?: string | null;
tournamentTitle?: string | null;
}

File: backend/TeamPilotApp/TeamPilot.Application/Dtos/team/TeamMatchDTO.cs
using TeamPilot.Application.Dtos.Tournament;

namespace TeamPilot.Application.Dtos.team
{
    public class TeamMatchDTO
    {
        public string MatchId { get; set; }
        public string TournamentId { get; set; }
        public string TournamentTitle { get; set; }
        public string MatchDate { get; set; }
        public List<TeamForTournamentDTO>? ParticipatingTeams { get; set; }
    }
}
